vsan/client.go

```diff
@@ -84,3 +87,78 @@ func (c *Client) VsanPerfQueryPerf(ctx context.Context, cluster *vimtypes.Manage
 	}
 	return res.Returnval, nil
 }
+
+// Creates the vsan object identities instance. This is to be queried from vsan health.
+var (
+	VsanQueryObjectIdentitiesInstance = vimtypes.ManagedObjectReference{
+		Type:  "VsanObjectSystem",
+		Value: "vsan-cluster-object-system",
+	}
+)
+
+// VsanQueryObjectIdentities return host uuid
+func (c *Client) VsanQueryObjectIdentities(ctx context.Context, cluster vimtypes.ManagedObjectReference) (*vsantypes.VsanObjectIdentityAndHealth, error) {
+	req := vsantypes.VsanQueryObjectIdentities{
+		This:    VsanQueryObjectIdentitiesInstance,
+		Cluster: cluster,
+	}
+
+	res, err := methods.VsanQueryObjectIdentities(ctx, c.serviceClient, &req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &res.Returnval, nil
+
+}
+
+// QueryVsanObjects Queries the vsan object
+func (c *Client) QueryVsanObjects(ctx context.Context, uuids []string, datacenter string, url string) (string, error) {
+
+	u, err := soap.ParseURL(url)
+	ctx = context.Background()
+	c1, err := govmomi.NewClient(ctx, u, true)
```

Comment:
why are we creating a new client?
function should use the `client` already defined in the func definition. `func (c *Client) QueryVsanObjects`.